nifi-toolkit/nifi-toolkit-cli/src/main/java/org/apache/nifi/toolkit/cli/impl/command/registry/bucket/UpdateBucketPolicy.java

```diff
@@ -0,0 +1,129 @@
+        String bucketId = getArg(properties, CommandOption.BUCKET_ID);
+
+        final String userNames = getArg(properties, CommandOption.USER_NAME_LIST);
+        final String userIds = getArg(properties, CommandOption.USER_ID_LIST);
+        final String groupNames = getArg(properties, CommandOption.GROUP_NAME_LIST);
+        final String groupIds = getArg(properties, CommandOption.GROUP_ID_LIST);
+
+        final String policyAction = getRequiredArg(properties, CommandOption.POLICY_ACTION);
+        final HashSet<String> permittedActions = new HashSet<>(Arrays.asList("read", "write", "delete"));
+        if (!permittedActions.contains(policyAction)) {
+            throw new IllegalArgumentException("Only read, write, delete actions permitted");
+        }
+        if (StringUtils.isBlank(bucketName) == StringUtils.isBlank(bucketId)) {
+            throw new IllegalArgumentException("Specify either bucket name or bucket id");
+        }
+        if (StringUtils.isBlank(bucketId)) {
+            final Optional<Bucket> optionalBucket = client.getBucketClient().getAll()
+                    .stream().filter(b -> bucketName.equals(b.getName())).findAny();
+            if (!optionalBucket.isPresent()) {
+                throw new IllegalArgumentException("Specified bucket does not exist");
+            }
+            bucketId = optionalBucket.get().getIdentifier();
+        } else {
+            try {
+                extendedClient.getBucketClient().get(bucketId);
+            } catch (NiFiRegistryException e) {
+                throw new IllegalArgumentException("Specified bucket does not exist");
+            }
+        }
+        AccessPolicy accessPolicy;
+        String resource = "/buckets/" + bucketId;
+        try {
+            accessPolicy = policiesClient.getAccessPolicy(policyAction, resource);
+        } catch (NiFiRegistryException e) {
+            accessPolicy = new AccessPolicy();
+            accessPolicy.setResource(resource);
+            accessPolicy.setAction(policyAction);
+        }
+        if (!StringUtils.isBlank(userNames) || !StringUtils.isBlank(userIds)) {
+            Set<Tenant> users = TenantHelper.selectExistingTenants(userNames,
+                    userIds, extendedClient.getTenantsClient().getUsers());
+            accessPolicy.setUsers(users);
+        }
+        if (!StringUtils.isBlank(groupNames) || !StringUtils.isBlank(groupIds)) {
+            Set<Tenant> groups = TenantHelper.selectExistingTenants(groupNames,
+                    groupIds, extendedClient.getTenantsClient().getUserGroups());
+            accessPolicy.setUserGroups(groups);
+        }
+        AccessPolicy updatedPolicy = StringUtils.isBlank(accessPolicy.getIdentifier())
+                ? policiesClient.createAccessPolicy(accessPolicy)
+                : policiesClient.updateAccessPolicy(accessPolicy);
+        println(updatedPolicy.getIdentifier());
+        return VoidResult.getInstance();
```

Comment:
If we want the result to be void, then we should wrap the println with `if (shouldPrint(properties)) { ... }`, this would make it only print when in interactive mode of the CLI, but not during non-interactive mode.

If you want the command to always return the policy id, then instead of println with Void, you would instead return a StringResult like:
`return new StringResult(identifier, getContext().isInteractive());`

Usually a create command returns the id as a string result and an update returns void, so since this command is a combo of either create or update, you could return different results depending whether you called create or update.